test/models/solid_queue/claimed_execution_test.rb

```diff
@@ -22,9 +22,7 @@ class SolidQueue::ClaimedExecutionTest < ActiveSupport::TestCase
     job = claimed_execution.job
 
     assert_difference -> { SolidQueue::ClaimedExecution.count } => -1, -> { SolidQueue::FailedExecution.count } => 1 do
-      assert_raises RuntimeError do
-        claimed_execution.perform
-      end
+      claimed_execution.perform
```

Comment:
This was added in https://github.com/rails/solid_queue/pull/373, and it's not clear to me that the new behaviour is correct 🤔  If the error reporting library doesn't rely on Rails error reporting, I think this might not work correctly.

Reply:
@rosa 

I'm not sure I fully understand your issue / question or possibly the impact of #373.  

Is your issue with the actual error handling and possibly missed / lost errors, this test is not valid with my changes and hence I'm no longer testing an important behavior, or what happens if a SQ user is depending on something other than Rails.error for their system wide Error manage mechanism?

hms

Reply:
@rosa 

I've done some digging into ActiveSupport::ExecutorWrapper and SolidQueue initialization and I think I understand your issue/concern. 

When an exception is raised and not rescued within a ExecutionWrapper.wrap block, it's caught and reported to the ActiveSupport.error_reporter.  Under normal circumstances, ActiveSupport.error_reporter and Rails.error_reporter are one and the same.  So, the code in the PR is fine.... under normal circumstances.

However, if directly or indirectly, the app_executor has been overridden and the overridden app_executor handles exception reporting without calling Rails.error_reporter, then the current implementation of the PR is effectively swallowing the exception.

If this summary of your concern is correct, I'll see what I have to do to address it.